crates/biome_cli/src/reporter/mod.rs

```diff
@@ -0,0 +1,206 @@
+pub(crate) mod terminal;
+
+use crate::cli_options::CliOptions;
+use crate::execute::Execution;
+use crate::CliDiagnostic;
+use biome_diagnostics::{Error, Severity};
+use std::io;
+use std::time::Duration;
+
+pub struct DiagnosticsPayload<'a> {
+    diagnostics: Vec<Error>,
+    verbose: bool,
+    diagnostic_level: Severity,
+    execution: &'a Execution,
+}
+
+/// A type that holds the result of the traversal
+#[derive(Debug, Default)]
+pub struct TraversalSummary {
+    changed: usize,
+    unchanged: usize,
+    duration: Duration,
+    errors: u32,
+    warnings: u32,
+    skipped: usize,
+    suggested_fixes_skipped: u32,
+    diagnostics_not_printed: u32,
+}
+
+impl TraversalSummary {
+    pub fn with_changed(mut self, value: usize) -> Self {
+        self.changed = value;
+        self
+    }
+    pub fn with_unchanged(mut self, value: usize) -> Self {
+        self.unchanged = value;
+        self
+    }
+    pub fn with_duration(mut self, value: Duration) -> Self {
+        self.duration = value;
+        self
+    }
+    pub fn with_errors(mut self, value: u32) -> Self {
+        self.errors = value;
+        self
+    }
+
+    pub fn with_warnings(mut self, value: u32) -> Self {
+        self.warnings = value;
+        self
+    }
+
+    pub fn with_skipped(mut self, value: usize) -> Self {
+        self.skipped = value;
+        self
+    }
+
+    pub fn with_suggested_fixes_skipped(mut self, value: u32) -> Self {
+        self.suggested_fixes_skipped = value;
+        self
+    }
+
+    pub fn with_diagnostics_not_printed(mut self, value: u32) -> Self {
+        self.diagnostics_not_printed = value;
+        self
+    }
+
+    pub fn changed(&self) -> usize {
```

Comment:
Dummy question: why not just make public the fields of `TraversalSummary`. This seems to be a container object.